src/app/app.routes.strategy.ts

```diff
@@ -41,6 +41,10 @@ interface RouteInfo {
 export class AppRouteReuseStrategy implements RouteReuseStrategy {
   private routeCache = new Map<string, RouteInfo>();
 
+  resetCache() {
```

Comment:
I tried locally to call destroy for the handled component and there are no more unsubscribe issues. It's a bit hacky, but it seems to work :)
```suggestion
  resetCache() {
    this.routeCache.forEach(value => {
      this.deactivateComponent(value.handle);
    });
    this.routeCache.clear();
  }

  private deactivateComponent(handle: DetachedRouteHandle): void {
    const componentRef: ComponentRef<any> = handle['componentRef'];
    if (componentRef) {
      componentRef.destroy();
    }
  }

```